File: Pythonleetcode/Leetcode409.py
class Solution(object):
    def longestPalindrome(self, s):
        ans=0
        a=(list(set(s)))
        max=0
        for i in a:
            b=s.count(i)
            if b%2==0:
                ans=ans+b

            else:
                ans=ans+b-1
                max=1
        ans=ans+max
        return ans






print(Solution().longestPalindrome("abbcccddddfffff"))
print(Solution().longestPalindrome("abccccdd"))

File: CppPractice/CppLeetcode/Leetcode409.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        map<char,int> theMap;
        bool isPlus=false;
        int ans = 0;
        for(char c:s)
        {
            if (theMap.find(c)==theMap.end())
            {
                theMap[c] = 1;
            }
            else
            {
                theMap[c] += 1;
            }
        }
        for(auto pair:theMap)
        {
            if (pair.second%2==1)
            {
                isPlus=true;
            }
            ans +=2*(pair.second/2);
        }
        if (isPlus)
        {
            ans++;
        }
        return ans;
        
    }
};